lib/iris/analysis/_scipy_interpolate.py

```diff
@@ -166,10 +207,79 @@ def __call__(self, xi, method=None):
                     raise ValueError("One of the requested xi is out of "
                                      "bounds in dimension %d" % i)
 
-        indices, norm_distances, out_of_bounds = self._find_indices(xi.T)
+        method = self.method if method is None else method
+        prepared = (xi_shape, method) + self._find_indices(xi.T)
+        if TIMINGS:
+            print('Prepare:', time.time() - t0)
+
+        if SPARSE and method == 'linear':
+            t0 = time.time()
+
+            xi_shape, method, indices, norm_distances, out_of_bounds = prepared
+
+            # Allocate arrays for describing the sparse matrix.
+            n_src_values_per_result_value = 2 ** ndim
+            n_result_values = len(indices[0])
+            n_non_zero = n_result_values * n_src_values_per_result_value
+            weights = np.ones(n_non_zero, dtype=norm_distances[0].dtype)
+            col_indices = np.empty(n_non_zero)
+            row_ptrs = np.arange(0, n_non_zero + n_src_values_per_result_value,
+                                 n_src_values_per_result_value)
+
+            corners = itertools.product(*[[(i, 1 - n), (i + 1, n)]
+                                          for i, n in zip(indices,
+                                                          norm_distances)])
+            shape = self.values.shape[:ndim]
+
+            for i, corner in enumerate(corners):
+                corner_indices = [ci for ci, cw in corner]
+                n_indices = np.ravel_multi_index(corner_indices, shape,
+                                                 mode='wrap')
+                col_indices[i::n_src_values_per_result_value] = n_indices
+                for ci, cw in corner:
+                    weights[i::n_src_values_per_result_value] *= cw
+
+            n_src_values = np.prod(map(len, self.grid))
+            sparse_matrix = csr_matrix((weights, col_indices, row_ptrs),
+                                       shape=(n_result_values, n_src_values))
+            if TIMINGS:
+                print('Convert:', time.time() - t0)
+
+            # XXX Hacky-Mc-Hack-Hack
+            prepared = (xi_shape, method, sparse_matrix, None, out_of_bounds)
+
+        return prepared
+
+    def interp_using_pre_computed_weights(self, computed_weights):
+        """
+        Do the interpolation, given pre-computed interpolation weights.
```

Comment:
You know what, the rest of the sentence isn't up to much either. How about...
`Perform the interpolation using pre-computed interpolation weights.`